src/NBitcoin/ChainBase.cs

```diff
@@ -4,15 +4,33 @@
 
 namespace NBitcoin
 {
+    /// <summary>Base implementation for chains.</summary>
     public abstract class ChainBase
     {
+        /// <summary>
+        /// Gets the chained block header given a block ID (hash).
+        /// </summary>
+        /// <param name="id">Block ID to retreive.</param>
+        /// <returns>The block.</returns>
```

Comment:
good for changing the comments to `chained block headers`, but you've forgotten some returns